### PR TITLE
Add logging for simulator actions

### PR DESCRIPTION
**Pull request by microbit-robert**

### CODE REVIEW

File: src/device/simulator.ts [hunk 1]
```diff
@@ -163,6 +175,10 @@ export class SimulatorDeviceConnection
       }
       case "request_flash": {
         this.emit(EVENT_REQUEST_FLASH);
+        this.logging.event({
+          type: "sim-user",
+          message: "start",
```
Comment (microbit-robert) [inline src/device/simulator.ts:180]: "start" or "play"?

**Comment by microbit-robert** [comment]
Is there a neater way of dealing with session storage in `simulator.ts`?

**Comment by microbit-robert** [comment]
We always call stop() on loading the page due to an effect running in `SimulatorActionBar.tsx` where we always start "out of sync".

File: src/device/simulator.ts (hunk 1 above)
Comment (microbit-matt-hillsdon) [reply src/device/simulator.ts:180]: Let's go for start.

Let's make sure we update the docs.

File: src/device/simulator.ts [hunk 2]
```diff
@@ -132,6 +132,18 @@ const initialDataLog = (): DataLog => ({
   data: [],
 });
 
+export type SensorsLogged = Record<string, boolean>;
+
+const SENSORS_LOGGED = "sensorsLogged";
+
+const getSensorsLogged = (): SensorsLogged => {
+  const sensorsLogged = sessionStorage.getItem(SENSORS_LOGGED);
```
Comment (microbit-matt-hillsdon) [inline src/device/simulator.ts:140]: I'm not sure this is worth the trouble of actual session storage.

File: src/device/simulator.ts [hunk 3]
```diff
@@ -286,11 +308,19 @@ export class SimulatorDeviceConnection
...
     prefixed.set(data, 3);
-    this.postMessage("radio_input", { data: prefixed });
+    this.postMessage(kind, { data: prefixed });
+    if (!this.sensorsLogged[kind]) {
+      this.logging.event({
+        type: "sim-user",
+        message: kind,
+      });
+      this.logSensor(kind);
+    }
```
Comment (microbit-matt-hillsdon) [inline src/device/simulator.ts:323]: Maybe dedup to a (private) logSensor call on `this`?

**Review by microbit-matt-hillsdon — CHANGES_REQUESTED** [review]
Minor tweaks.

File: src/device/simulator.ts (hunk 3 above)
Comment (microbit-robert) [reply src/device/simulator.ts]: [`2dc5220`](https://github.com/microbit-foundation/python-editor-v3/pull/1021/commits/2dc52203f09a15db813d0952316697974d8ce5d4).

File: src/device/simulator.ts (hunk 2 above)
Comment (microbit-robert) [reply src/device/simulator.ts]: Removed the use of session storage -> [`2dc5220`](https://github.com/microbit-foundation/python-editor-v3/pull/1021/commits/2dc52203f09a15db813d0952316697974d8ce5d4).

**Comment by microbit-matt-hillsdon** [comment]
> We always call stop() on loading the page due to an effect running in `SimulatorActionBar.tsx` where we always start "out of sync".

Let's discuss this one. I'm not sure we want to log it anyway as it's not a user action.

**Comment by microbit-robert** [comment]
> > We always call stop() on loading the page due to an effect running in `SimulatorActionBar.tsx` where we always start "out of sync".
> 
> Let's discuss this one. I'm not sure we want to log it anyway as it's not a user action.

Changes as discussed -> [`12f7262`](https://github.com/microbit-foundation/python-editor-v3/pull/1021/commits/12f726292649c0069ffc6382de6cb0c0695889bc).

**Review by microbit-matt-hillsdon — APPROVED** [review]
Looks good, thanks.

**Pull request merged**